homeassistant/config.py

```diff
@@ -521,16 +524,35 @@ def set_time_zone(time_zone_str: Optional[str]) -> None:
         time_zone = date_util.get_time_zone(time_zone_str)
 
         if time_zone:
+            hac.config_source = 'yaml'
             hac.time_zone = time_zone
             date_util.set_default_time_zone(time_zone)
         else:
             _LOGGER.error("Received invalid time zone %s", time_zone_str)
 
+    store = hass.helpers.storage.Store(CORE_STORAGE_VERSION, CORE_STORAGE_KEY,
```

Comment:
Should this happen here in `config.py`, or should it be done somewhere else?

Reply:
At least should put in a separate function